library/src/main/java/com/datatorrent/lib/window/TriggerOption.java

```diff
@@ -0,0 +1,55 @@
+package com.datatorrent.lib.window;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ *
+ */
+public class TriggerOption
+{
+
+  public static enum WatermarkOpt
+  {
+    BEFORE,
+    AFTER
+  }
+
+  WatermarkOpt watermarkOpt = WatermarkOpt.BEFORE;
+
+  private List<Object[]> size = new LinkedList<>();
+
+  public static class TriggerOptionBuilder
+  {
+
+    public static TriggerOption every(long quantity, Quantification.Unit unit)
```

Comment:
We will change the interfaces with quantity and unit to something different. Stay tuned.